### ISSUE TITLE
Java Driver: ReqlExpr AND / OR inconsistency

### ISSUE THREAD

**Issue by bchavez** [rethinkdb/rethinkdb#5547]
An issue came up in the **C# driver** regarding `ReqlExpr` `And` and `Or` method signatures. https://github.com/bchavez/RethinkDb.Driver/issues/36

It also appears the **Java driver** exhibits the same issue, specifically:

**Java Driver:** [`ReqlExpr.java`](https://github.com/rethinkdb/rethinkdb/blob/next/drivers/java/src/main/java/com/rethinkdb/gen/ast/ReqlExpr.java#L1746)
```
public Or or(Object exprA, Object... exprs) {
    ....
}
public And and(Object... exprs) {
    ....
}
```

**C# Driver:** [`ReqlExpr.cs`](https://github.com/bchavez/RethinkDb.Driver/blob/master/Source/RethinkDb.Driver/Generated/Ast/ReqlExpr.cs#L5807)
```
public Or Or(Object exprA, params object[] exprs) {
    ....
}
public And And(params object[] exprs){
    ....
}
```

@VeXocide mentioned in slack that `r.and()` and `r.or()` were recently changed to accept zero or more arguments referencing: https://github.com/rethinkdb/docs/issues/944

cc: @quentez

**Comment by danielmewes** [comment]
Thanks for checking the Java driver and opening the issue here @bchavez .

Looks like `Or` should get the same signature as `And`. 
Pinging @deontologician .

**Comment by danielmewes** [comment]
Note that the 0-argument version of `r.and` and `r.or` is not available in RethinkDB 2.2.x, but is an upcoming RethinkDB 2.3 feature. So this is probably not a big issue right now.

**Comment by danielmewes** [comment]
A fix is in review 3546, and commit https://github.com/rethinkdb/rethinkdb/commit/762875067d0fd951c6c4ff97f1ed32365d9e7a0c .

@bchavez Note that this required a change to the AstSubclass template, as you can see here: https://github.com/rethinkdb/rethinkdb/commit/762875067d0fd951c6c4ff97f1ed32365d9e7a0c#diff-9952793414fd4dee1b383f0bbd945f7fL43

**Comment by bchavez** [comment]
Thanks very much @danielmewes . Also, thanks for the heads up on the `AstSubclass` template change. I'll try to get this ported later this weekend or early next week.

Also, just out of curiosity, where did `.wait_()`, `.reconfigure()`, and `.rebalance()` go? Were they dupes in `TopLevel`?

**Comment by danielmewes** [comment]
@bchavez: Those terms are no longer valid on the top level: https://github.com/rethinkdb/rethinkdb/issues/4382

**Comment by danielmewes** [comment]
This is fixed in `next` together by 95dfdb2, together with support for the new `union` `interleave` optarg and complete support for the new `fold` command.
Will be released with RethinkDB 2.3.0.

**Comment by bchavez** [comment]
@danielmewes Thanks very much for the info and diffs. It was very helpful porting downstream. The regenerated C# driver AST now has the new `fold` term and support for `union:interleave`. https://github.com/bchavez/RethinkDb.Driver/commit/94caa94a008651532b59560229c859041e5e608b

I'm probably going to hold off on re-generating unit tests until 2.3.0 is released since it looks like we have a new `fold.yaml` test group.
